zenodo-master/docs/api/config.rst
..
    This file is part of Zenodo.
    Copyright (C) 2015 CERN.

    Zenodo is free software; you can redistribute it
    and/or modify it under the terms of the GNU General Public License as
    published by the Free Software Foundation; either version 2 of the
    License, or (at your option) any later version.

    Zenodo is distributed in the hope that it will be
    useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Zenodo; if not, write to the
    Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
    MA 02111-1307, USA.

    In applying this license, CERN does not
    waive the privileges and immunities granted to it by virtue of its status
    as an Intergovernmental Organization or submit itself to any jurisdiction.


Base Configuration
------------------

.. automodule:: zenodo.config
   :members:
   :undoc-members:


bento-master/CHANGELOG.md
# Changelog

This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [0.11.1](https://pypi.org/project/bento-cli/0.11.1/) - 2020-04-28

### Fixed

- Upgrade notice now prints to stderr, to avoid affecting JSON output
- Tools with file-based configuration (e.g. `sgrep`) will now have
  their caches invalidated when these configuration files change

## [0.11.0](https://pypi.org/project/bento-cli/0.11.0/) - 2020-04-17

### Added

- Added `gosec`, an open-source tool to detect security vulnerabilities in
  Go projects; to use, run `bento enable tool gosec`
- Both r2c check registries and custom check definitions can now be used with
  the `sgrep` tool; to use a r2c check registry, run
  ```
  BENTO_REGISTRY=r/<registry> bento check -t sgrep ...
  ```
  to use a custom sgrep.live share link, run
  ```
  BENTO_REGISTRY=<share code> bento check -t sgrep ...
  ```
  with your three-character share code (these are the last three characters
  of your sgrep.live share link, e.g. `xQn` from `https://sgrep.live/xQn`)
- Bento can now run all tools within an enclosing Docker container; when
  running Bento within a Docker container, use
  ```
  BENTO_REMOTE_DOCKER=1 bento ...
  ```

### Removed

- Removed the experimental `python-taint` tool
- Removed dependency on `r2c-lib`

### Changed

- The `r2c.boto3` checks are now enabled by default; to disable them on
  a project, run `bento disable tool r2c.boto3`

## [0.10.2](https://pypi.org/project/bento-cli/0.10.2/) - 2020-04-06

### Fixed

- Fixed an issue where the `r2c.registry.latest` tool would fail to run
  in a GitHub Action or other environment with a remote Docker daemon,
  as indicated with the `R2C_USE_REMOTE_DOCKER` environment variable.

## [0.10.1](https://pypi.org/project/bento-cli/0.10.1/) - 2020-03-04

### Added

- Add `ci_provider` field to telemetry and update
  [PRIVACY.md](https://github.com/returntocorp/bento/blob/master/PRIVACY.md) accordingly

### Fixed

- Fix a regression (since 0.8.0) where root paths were not properly ignored
- Fix a regression (since 0.10.0) where `init` failed to offer to update the user's
  global git ignore file
- If .bentoignore is missing, render a clear error message to the user

## [0.10.0](https://pypi.org/project/bento-cli/0.10.0/) - 2020-02-26

### Added

- Added `r2c.registry.latest`, an experimental feature that pulls the latest checks from r2c's check registry
- `bento enable ci` to setup the [Bento GitHub Action](https://github.com/marketplace/actions/bento-check) for applicable projects
- `bento init` now prompts to setup the GitHub Action (when applicable)
- Dlint is enabled by default

### Fixed

- Use shebang detection for shell and python files during `bento init`
- jinja tool only installs if it detects python and html/jinja files
- Fix jinja tool hyperlinks
- Exit when a configured tool cannot be found instead of failing silently

## [0.9.1](https://pypi.org/project/bento-cli/0.9.1/) - 2020-02-14

### Fixed

- ShellCheck detect shell files robustly

## [0.9.0](https://pypi.org/project/bento-cli/0.9.0/) - 2020-02-13

### Added

- Python tools now scan files that don't end in `.py` but have python shebang
- Added ReDoS check via Dlint. The check is off by default. Enable the Dlint tool to run this check: `bento enable tool dlint`.
- Jinja checks added and turned on by default

See [https://bento.dev/checks](https://bento.dev/checks) for more information on new checks

### Fixed

- Bento iterates over files in batches if number of files to check exceeds OS argmax limit

## [0.8.2](https://pypi.org/project/bento-cli/0.8.2/) - 2020-02-05

### Fixed

- `bento check` works with repos with no previous commits
- Verify a path is a file before checking if it contains a shell shebang (shellcheck tool)
- Correctly install autorun even if git-hooks subdirectory is non-existent

## [0.8.1](https://pypi.org/project/bento-cli/0.8.1/) - 2020-01-30

### Fixed

- Perfomance improvements when running in large projects
- Additional human readable check_ids for bandit

## [0.8.0](https://pypi.org/project/bento-cli/0.8.0/) - 2020-01-24

This release represents a major shift in Bento's default behavior: It emphasizes an incremental
and personal, rather than team-wide, workflow that makes Bento a smaller commitment to use:

1. Other project contributors won’t see Bento files or have their workflows changed.
2. You no longer need to manually run Bento. After initialization Bento will automatically
   check for issues in your code as you commit, analyzing only the files that have changed.
3. You won’t see a project’s old issues (tech debt) during initialization. To view them,
   run

   ```bash
   bento check --all
   ```

### Migration

Project configurations have changed in version 0.8. In order to migrate a project from
version 0.7 or earlier:

- Ensure Bento has been upgraded using `pip3 install --upgrade bento-cli`. Run `bento --version`
  to validate Bento’s version.
- `rm -r .bento*` (Use `git rm` if you have previously added Bento files to source control).
- Run `bento init` in the project root.

### New requirements

- Docker must be installed, and the docker client running, to use Bento.

### Changed

How you use Bento has changed significantly in 0.8.

#### Usage changes

- `bento check` will now only check staged changes.
  - Use `bento check --all` to check the entire project.
- `bento archive` will archive findings due to staged diff:

  - Use `bento archive --all` to archive all findings in a project.

#### Other changes

- `hadolint` and `shellcheck` are now enabled by default.
- Messages for findings are no longer truncated.
- `bento init` will now install an empty configuration on a project it can not identify;
  tools may then be manually enabled using `bento enable tool TOOL`. Use `bento enable tool --help`
  to list tools.
- Virtual environments for Python tools are now installed in your home directory, instead of in your
  project directories.

repos/Hermetica-master/hermetica/cli.py
#! /usr/bin/env python3
# -*- encoding: utf-8 -*-
"""
main.py
hermetica main script
"""
__author__ = 'Yoshiya Ito <myon53@gmail.com>'
__version__ = '1.0.0'
__date__ = '2018-04-24'
import os
import click
from hermetica.scaffold.app import App
from hermetica.scaffold.config import Config
from hermetica.scaffold.wsgi import WSGI
from hermetica.scaffold.api import API
from hermetica.scaffold.model import Model
from hermetica.scaffold.test import Test
from hermetica.scaffold.docker import Docker
from hermetica.scaffold.extension import Extension
from hermetica.scaffold.pipfile import Pipfile
from hermetica.scaffold.decorator import Decorator

@click.group()
def main():
    pass

@main.command()
@click.option('--api', default='restful', type=click.Choice(['restful', 'decorator', 'class']), help='Flask-Restful or Flask decorator or methodview')
@click.option('--db', default=None, type=click.Choice(['sqlalchemy', 'mongoengine']), help='SQLAlchemy or Mongoengine or None')
@click.option('--decorator', default=False, is_flag=True, help='create decorator or None')
@click.option('--redis', default=False, is_flag=True, flag_value='redis', help='using Redis or None')
@click.option('--docker', default=False, is_flag=True, flag_value='docker', help='using container')
def init(api, db, decorator, redis, docker):
    """ initialize your flask app
    """
    dirs = ['./app/', './test/', './config/', './app/api/v1/', './app/models/']
    for dir in dirs:
        if os.path.exists(os.path.dirname(dir)):
            click.echo('[WARNING] directory {} is already exists, skip to create this directory'.format(dir))
            continue
        os.makedirs(os.path.dirname(dir))

    app = App(db=db, redis=redis, api=api)
    pipfile = Pipfile(db=db, redis=redis)
    wsgi = WSGI(db=db)
    config = Config(db=db, redis=redis)
    extension = Extension(db=db, redis=redis)
    test = Test(db=db, name='root')
    api = API(api=api, name='root')
    decorator = Decorator(name='root')

    with open('./Pipfile', 'w') as f:
        f.write(pipfile.create_pipfile())
    with open('wsgi.py', 'w') as f:
        f.write(wsgi.create_wsgi())

    with open('app/__init__.py', 'w') as f:
        f.write(app.create_app__init__())
    with open('app/extensions.py', 'w') as f:
        f.write(extension.create_extensions())
    if decorator:
        with open('app/decorators.py', 'w') as f:
            f.write(decorator.create_decorators())

    with open('config/__init__.py', 'w') as f:
        f.write(config.create_config(name='config', env='test'))
    with open('config/development.py', 'w') as f:
        f.write(config.create_config(name='development', env='development'))
    with open('config/production.py', 'w') as f:
        f.write(config.create_config(name='production', env='production'))

    if docker:
        docker = Docker(db=db, redis=redis)
        with open('Dockerfile', 'w') as f:
            f.write(docker.create_dockerfile())
        with open('docker-compose.yml', 'w') as f:
            f.write(docker.create_docker_compose_yml())

    with open('test/__init__.py', 'w') as f:
        f.write(test.create__init__())
    with open('nose.cfg', 'w') as f:
        f.write(test.create_nose_cfg())

    with open('app/api/__init__.py', 'w') as f:
        f.write(api.create__init__())
    with open('app/api/v1/__init__.py', 'w') as f:
        pass
    with open('app/api/v1/root.py', 'w') as f:
        f.write(api.create_api())

    if db:
        model = Model(db=db, name='root')
        with open('app/models/__init__.py', 'w') as f:
            f.write(model.create__init__())
        with open('app/models/root.py', 'w') as f:
            f.write(model.create_model())

@main.command()
@click.argument('name', type=str, required=True)
@click.option('--api', default='restful', type=click.Choice(['restful', 'decorator', 'class']), help='Flask-Restful or Flask decorator or methodview')
@click.option('--version', default='v1', help='API version')
def api(name, api, version):
    """ create api
    """
    path = 'app/api/{}/{}.py'.format(version, name)
    api = API(api=api, name=name)
    with open(path, 'w') as f:
        f.write(api.create_api())

@main.command()
@click.argument('name', type=str, required=True)
@click.option('--db', default='sqlalchemy', type=click.Choice(['sqlalchemy', 'mongoengine']), help='SQLAlchemy or Mongoengine or None')
def model(name, db):
    """ create model
    """
    path = 'app/models/{}.py'.format(name)
    model = Model(db=db, name=name)
    with open(path, 'w') as f:
        f.write(model.create_model())

@main.command()
@click.argument('name', type=str, required=True)
def decorator(name):
    """ create decorator
    """
    decorator = Decorator(name=name)
    with open('app/decorators.py', 'a') as f:
        f.write(decorator.create_decorator())
